python_modules/libraries/dagster-teradata/dagster_teradata/resources.py

```diff
@@ -0,0 +1,464 @@
+from contextlib import closing, contextmanager
+from datetime import datetime
+from textwrap import dedent
+from typing import Any, Iterator, List, Mapping, Optional, Sequence, Union
+
+import dagster._check as check
+import teradatasql
+from dagster import (
+    ConfigurableResource,
+    IAttachDifferentObjectToOpContext,
+    get_dagster_logger,
+    resource,
+)
+from dagster._annotations import public
+from dagster._core.definitions.resource_definition import dagster_maintained_resource
+from dagster._core.storage.event_log.sql_event_log import SqlDbConnection
+from dagster._utils.cached_method import cached_method
+from dagster_aws.s3 import S3Resource
+from pydantic import Field
+
+from dagster_teradata import constants
+from dagster_teradata.teradata_compute_cluster_manager import TeradataComputeClusterManager
+
+
+class TeradataResource(ConfigurableResource, IAttachDifferentObjectToOpContext):
```

Comment:
I feel, this resource can be divided into three parts. One for Handling normal query execution, second one for DataTransfer and other for ComputeCluster. Please think on this

Reply:
Sure, will check the possibility